code/modules/medical_emergencies/medical_emergency.dm

```diff
@@ -0,0 +1,23 @@
+/datum/medical_effect
+	var/name = "Medical Emergency"
+	var/description = "Brief summary of effect"
+	var/reccomended_treatment = "Treatment method"
+	var/stage = 1
+	var/current_tick = 0
+	var/stage_advance_tick = 10
+	var/stage_advance_prob = 100
+	var/max_stage = 3
+
+/datum/medical_effect/proc/on_granting(var/mob/living/carbon/M)
+	return
+
+/datum/medical_effect/process(var/mob/living/carbon/M)
+	current_tick++
+	if(current_tick == stage_advance_tick && stage != max_stage)
+		if(prob(stage_advance_prob))
+			stage++
+		current_tick = 0
```

Comment:
Alright, on Goon...it's just a flat probability chance to advance a stage (unless I'm reading the code wrong).

If you have the right reagents to cure an effect (or whatever other condition), then the effect rolls for a chance to become "remissive". When it's remissive, it starts rolling for a chance to go from its current stage back 1 stage--once it hits stage 0, it's removed/cured.

This may sound like a pointlessly fine thing, but it helps Adds to the tension of crit and make it a bit more lethal than "lol, I rolled for succesful cure--medical effect removed entirely".